src/components/v2/V2Project/V2ManageTokensSection/V2RedeemModal.tsx

```diff
@@ -244,15 +255,47 @@ export default function V2RedeemModal({
               onChange={val => setRedeemAmount(val)}
             />
           </Form>
-          {primaryTerminalCurrentOverflow?.gt(0) ? (
+          {totalSupplyExceeded ? (
+            <div style={{ color: colors.text.failure, marginTop: 20 }}>
+              <Trans>
+                You have exceeded the total token supply of{' '}
+                <strong>{formatWad(totalTokenSupply, { precision: 4 })}</strong>
+              </Trans>
+            </div>
+          ) : null}
+          {primaryTerminalCurrentOverflow?.gt(0) &&
+          !totalSupplyExceeded &&
+          minReturnedTokens?.gt(0) ? (
             <div style={{ fontWeight: 500, marginTop: 20 }}>
-              {distributionLimitCurrency?.eq(V2_CURRENCY_USD) ? (
-                <Trans>
-                  You will receive minimum {minReturnedTokensFormatted} ETH
-                </Trans>
-              ) : (
-                <Trans>You will receive {minReturnedTokensFormatted} ETH</Trans>
-              )}
+              <>
```

Comment:
**suggestion** this whole block is hard to follow, and there's a bit of duplication.

Could you find a way to clean it up? Happy to offer a patch tomorrow if I'm being too vague.

Reply:
Can understand your point, however, since we want these translated, I don't think there's any way to avoid writing out 4 blocks like this. I've added a comment to clarify, but not sure there's anything else I can do